pandas/io/formats/excel.py

```diff
@@ -356,7 +356,22 @@ def __init__(self, df, na_rep='', float_format=None, cols=None,
             self.styler = None
         self.df = df
         if cols is not None:
-            self.df = df.loc[:, cols]
+
+            # all missing, raise
+            if not len(Index(cols) & df.columns):
+                raise KeyError(
+                    "passes columns are not ALL present dataframe")
+
+            # deprecatedin gh-17295
+            # 1 missing is ok (for now)
+            if len(Index(cols) & df.columns) != len(cols):
+                warnings.warn(
+                    "columns must be a subset of the "
+                    "dataframe columns; this will raise "
+                    "a KeyError in the future",
```

Comment:
When you only see that error message, it is not directly clear that on the the names you passed in `columns` is missing (you directly say the result). Can you clarify that? 
Maybe like "Not all names specified in 'columns' are found(or present); this will raise a KeyError in the future"

Reply:
sure